tests/DPDK_Flow_Dump_with_del_add.py

```diff
@@ -0,0 +1,173 @@
+                    self.fail(f"Failed to add table entry {match_action}")
+      
+        # Get dump entries table
+        dump_table = p4rt_ctl.p4rt_ctl_dump_entities(self.config_data['switch'])
+        # Verify each entry
+        if len(dump_table[1:-1]) != len(self.config_data["flow_dump_table"]):
+            log.failed(f"The table {dump_table[1:-1]} has different entry than definition")
+            self.result.addFailure(self, sys.exc_info())
+            self.fail(f"Failed to verify flow dump entries\"")
+        else:
+            for each in dump_table[1:-1]:
+                entry = each.strip()
+                if entry in self.config_data["flow_dump_table"]:
+                    log.passed(f"The entry \"{entry}\" in flow dump is verified")
+                else:
+                    log.failed(f"The entry \"{entry}\" in flow dump is not verified")
+                    self.result.addFailure(self, sys.exc_info())
+                    self.fail(f"Failed to verify entry \"{entry}\"")
+                    
+        # Delete entry 
+        log.info("Delete table entry and then verify flow dump")      
+        for table in self.config_data["table"]:
+            log.info(f"Deleting {table['description']} rules")
+            for del_action in table["del_action"]:
+                p4rt_ctl.p4rt_ctl_del_entry(table["switch"], table["name"], del_action)
+        
+        log.info("Verify flow dump table after deleting rules")       
+        dump_table = p4rt_ctl.p4rt_ctl_dump_entities(self.config_data['switch'])
+        if (len(dump_table[1:-1])) != 0 :
+            log.info("The table is {dump_table[1:-1])} after deletion")
+            log.failed(f"The table it not emty after entry deletion")
+            self.result.addFailure(self, sys.exc_info())
+            self.fail(f"Failed to verify \"{dump_table[1:-1]}\" after deletion")
+        else:
+           log.passed("No entry found in flow table after deletion")
+           
+        # Add back rules for l3_exact match
+        log.info("Add back rules for l3_exact match")
+        for table in self.config_data["table"]:
+
+            log.info(f"Scenario : l3 exact match : {table['description']}")
+            log.info(f"Adding {table['description']} rules")
+            for match_action in table["match_action"]:
+                if not p4rt_ctl.p4rt_ctl_add_entry(
+                    table["switch"], table["name"], match_action
+                ):
+                    self.result.addFailure(self, sys.exc_info())
+                    self.fail(f"Failed to add table entry {match_action}")
+        
+        log.info("Verify flow dump table after adding back rules")
+        # Verify each entry        
+        dump_table = p4rt_ctl.p4rt_ctl_dump_entities(self.config_data['switch'])
+        if len(dump_table[1:-1]) != len(self.config_data["flow_dump_table"]):
+            log.failed(f"The table {dump_table[1:-1]} has different entry than definition")
```

Comment:
same as above

Reply:
The statements have a variables {entry}. The f is needed